src/libiotc/iotc_user_sub_call_wrapper.c

```diff
@@ -73,7 +73,7 @@ iotc_state_t iotc_user_sub_call_wrapper(void* context, void* data,
           msg->publish.content ? msg->publish.content->data_ptr : NULL;
       params.message.temporary_payload_data_length =
           msg->publish.content ? msg->publish.content->length : 0;
-      params.message.topic = (const char*)sub_data->subscribe.topic;
+      params.message.topic = (const char*)msg->publish.topic_name->data_ptr;
```

Comment:
This puts the published topic into the callback handler so callbacks looking for wildcards can get the subfolder topic